### PR TITLE
MGMT-7251: Add minimal-initrd endpoint

### PR DESCRIPTION
**Pull request by djzager**
# Assisted Pull Request

## Description

Make it possible for the assisted-image-service to ask us for a cluster's minimal initial ramdisk.

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @carbonin 
/cc @

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by carbonin — COMMENTED** [review]
Maybe change the JIRA id in the cherry-picked commits.
I don't think it makes much sense to link it to an issue that's now closed.

File: internal/bminventory/inventory.go
Comment (carbonin) [reply internal/bminventory/inventory.go]: Ah, you fixed this in a later commit.
I was reviewing commit-by-commit. I'll just resolve this.

File: swagger.yaml
Comment (djzager) [reply swagger.yaml:494]: I like that.

**Review by flaper87 — COMMENTED** [review]
FWIW, I took a look at this PR yday but preferred to wait for @carbonin's review before commenting. It looks good overall and I think Nick's comment w.r.t paths makes sense.

Happy to re-review/approve once that's done.

File: swagger.yaml
Comment (carbonin) [inline swagger.yaml:3762]: This should also have `urlAuth: []` like the files download for when we run with the operator.

Comment (djzager) [reply swagger.yaml:3706]: Added.

**Review by flaper87 — APPROVED** [review]
:pizza:

File: swagger.yaml
Comment (avishayt) [inline swagger.yaml:4092]:
```suggestion
      operationId: DownloadMinimalInitrd
```

File: swagger.yaml
Comment (avishayt) [inline swagger.yaml:4083]: Doesn't need to be in this PR, but do we also need an API for getting a presigned URL?

File: internal/bminventory/inventory_test.go
Comment (flaper87) [inline internal/bminventory/inventory_test.go:5808]: nit: almost worth putting this pull secret in a global var, considering how that it's being copy/pasted throughout this test

File: internal/bminventory/inventory_test.go
Comment (flaper87) [inline internal/bminventory/inventory_test.go:5819]:
```suggestion
```

File: internal/bminventory/inventory_test.go
Comment (flaper87) [inline internal/bminventory/inventory_test.go:5836]:
```suggestion
```

**Review by flaper87 — COMMENTED** [review]
Looks good! Just have some nits to clean up commented code. Focused mostly on the tests, as requested. 

I do think we should do the rename that Avishay suggested in this PR. Other than that, looks good to go.

File: swagger.yaml
Comment (carbonin) [reply swagger.yaml:4083]: Not sure I understand. This is realistically only going to be used by the image service. And it will use whatever authentication it's provided by the user in the image download url.

File: internal/bminventory/inventory_test.go
Comment (carbonin) [inline internal/bminventory/inventory_test.go:5846]: Think this should be removed, right?

File: pkg/auth/auth_assisted_service_mock_test.go
Comment (carbonin) [reply pkg/auth/auth_assisted_service_mock_test.go:60]: Was this something you looked into @djzager ?

Comment (djzager) [reply pkg/auth/auth_assisted_service_mock_test.go:60]: Yeah. As far as I could tell the only thing this change addressed was to make the fake inventory satisfy the interface. I could not find a scenario where `GetMinimalInitrd` was called using this fake inventory.

**Review by carbonin — APPROVED** [review]

**Pull request merged**
